CONTRIBUTING.md

```diff
@@ -127,7 +127,39 @@ Run `rush rebuild` from anywhere in the repo to rebuild all projects from scratc
 
 Run `rush build -t <packagename>` to build a single project, and all local projects that it depends on. You can pass `-t` multiple times to build multiple projects. This works for `rush rebuild` as well. Keep in mind that Rush refers to packages by their full names, so packages will be named something like `@azure/<servicename>`.
 
-By default, Rush only displays things written to `STDERR`. If you want to see the full output, pass `--verbose` to any of the build commands. This is also true for script commands (see below).
+By default, Rush only displays things written to `STDERR`. If you want to see the full output, pass `--verbose` to any of the build commands.
+
+### Testing
+
+If you want to run the tests of a specific project, go to that project's folder and execute `rushx test`. All of the tests will automatically run both in NodeJS and in the browser. To target these environments individually, you can run `rushx test:node` and `rushx test:browser`.
+
+By default, these npm scripts run previously recorded tests. The recordings have been generated by using a custom recording library called [test-utils-recorder](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md). We will examine how to run recorded tests and live tests in the following sections.
+
+#### Recorded tests
+
+Most of the tests in our projects run in playback mode by default, i.e they make no network requests to the real services. For http requests made in each test case, there is a recorded response that reproduces the service behavior. The readme file in the `test` folder of each package will indicate wether the package uses recorded tests or not.
+
+#### Live tests
+
+In case you want to run the tests of the packages that use our [recorder](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md) against live resources, you will continue to run the `test` npm script, but you will need to ensure that:
+
+- The environment variable `TEST_MODE` is set to `live`.
+- You must have had previously created the necessary Azure resources needed by the tests you want to execute.
+- You must also set the appropriate environment variables to point to these resources.
+
+The readme file in the `test` folder of each package lists the Azure resources and the environment variables needed.
+
+> Note: Our projects use [dotenv](https://www.npmjs.com/package/dotenv) to make it easier to work with environment variables.
```

Comment:
I still see benefit in calling out the .env file. My previous comments were mainly for calling out other ways to set env variables. So, am fine hinting at it in a short sentence.

```suggestion
> Note: Our projects use [dotenv](https://www.npmjs.com/package/dotenv) to make it easier to work with environment variables. So, you can make use of .env files to set environment variables.
```

Reply:
What about:

> Note: Though you can set the environment variables as you normally would, our projects use [dotenv](https://www.npmjs.com/package/dotenv) to allow you to use `.env` files to set environment variables. Find the `sample.env` file closest to the project you want to test to have a template of the expected `.env` file already placed where the tests expect it to be.

This makes me wonder: Should we standardize how we use `dotenv`? I am seeing that not every package places the `.env` file at the same level.

Reply:
Yes, we should standardize it

Reply:
I made this issue: https://github.com/Azure/azure-sdk-for-js/issues/8526​​​​​​​